src/Build.UnitTests/BinaryLogger_Tests.cs

```diff
@@ -138,6 +140,31 @@ public void BinaryLoggerShouldSupportFilePathExplicitParameter()
             ObjectModelHelpers.BuildProjectExpectSuccess(s_testProject, binaryLogger);
         }
 
+        [Fact]
+        public void BinaryLoggerShouldEmbedFilesViaTaskOutput()
+        {
+            using var buildManager = new BuildManager();
+            var binaryLogger = new BinaryLogger()
+            {
+                Parameters = $"LogFile={_logFile}",
+                CollectProjectImports = BinaryLogger.ProjectImportsCollectionMode.ZipFile,
+            };
+            var testProject = @"
+<Project>
+    <Target Name=""Build"">
+        <WriteLinesToFile File=""./testtaskoutputfile.txt"" Lines=""abc;def;ghi""/>
+        <CreateItem Include=""./testtaskoutputfile.txt"">
```

Comment:
I imagine these slashes don't work on windows?

Reply:
I think the convention is to use the `\` on all OSs (and MSBuild will convert correctly where needed).

Reply:
Will look at this later today - don't want to use build infrastructure to potentially back and forth on test success.